src/reports/components/report-sections/report-collapsible-container.tsx

```diff
@@ -26,7 +27,10 @@ const ReportCollapsibleContainer = NamedFC<ReportCollapsibleContainerProps>(
 
         return (
             <div className={outerDivClassName}>
-                <div className="title-container" role="heading" aria-level={headingLevel}>
+                <HeadingElementForLevel
+                    headingLevel={headingLevel as HeadingLevel}
```

Comment:
Rather than using a type assertion here, it might be better to update the type of `headingLevel` in `ReportCollapsibleContainerProps` to be `HeadingLevel`. I believe there are a few other places in our props where `headingLevel: number` would make more sense as `headingLevel: HeadingLevel` with this change, as well 🙂

Reply:
@jalkire thanks a lot for looking at this! 

You are right I overlooked the css change, sorry about that. COuld you guide me to the use of the report-collapsible-container? I am not 100% sure I am looking at the right element. For the issue you pointed out, I believe I found a solution for the issue, I have added a new commit. 

I can see that there is also an issue with the styling of the heading in how-to-test, it used to not be styled as a heading and now has the heading styling. I just want to verrify that this is really how it is supposed to be (since I am completly new here)?  
old:

![old_how to test](https://user-images.githubusercontent.com/13312252/156052066-64116e08-3c1e-40d7-bcff-cba985786378.PNG)

new: 
![new_how to test](https://user-images.githubusercontent.com/13312252/156052136-1e12b3a7-2f47-4f81-b1cd-3e598b6373e8.PNG)

Reply:
@jalkire one more question, the snapshot test src/tests/unit/tests/reports/components/report-sections/report-collapsible-container.test.tsx:35:45 is failing as it is expecing a <div> element but getting a </HeadingElementForLevel> now, but I can't seem to find the source of the error. With my very limited knowledge of snapshot testing I assume a snapshot needs to be updated somewhere but can't find the correct one. Can you guide me? Thanks again!

Reply:
Great, thank you for those updates! To your questions:

- For the use of report-collapsible-container, we use that in the html reports that the extension exports. The easiest way to see it is by clicking "Export result" from a FastPass in the extension--the parts of the downloaded html report that expand/collapse on click are in the `ReportCollapsibleContainer`. 
- For the snapshots, if you add `-u` after the test command that will update the snapshots. Then you can confirm that the changes make sense 🙂 
- For the tab stops how to test heading, I would default to leaving it as-is, but tagging @ferBonnin in case the heading style should be changed.

Reply:
thanks for this contributions @majaokholm! For the Tab stops _How to test_ Heading, I recommend we have it as an **H2**, so its the same as the _why it matters_ headings 🙂

Reply:
@jalkire Thanks again for the feedback! But I am afraid I need to ask for a bit of help once more; to fix the last styling differences I need to update css for the report-collapsible-container (because the default browser styling for h4 elements is being applied now), but I can’t figure out which components uses which stylesheet, or how to add a new one. It should be so simple but I spend so much time trying to resolve this that I figured out it is time to ask for help. Can you please guide me once more? Thanks again!

Reply:
Sure thing! Unfortunately this particular styling looks a little messy on our end--tab-stops-failed-instance-section.scss, result-section.scss, automated-checks-report.scss, and collapsible-url-result-section.scss all set properties on title-container in report-collapsible-container.tsx. I might be missing something myself, but you're probably best off applying your fix to all of those.

This somewhat relates to the last failing PR check, e2e-report-tests--once you get that css updated, you can get the test passing by running `yarn build:package:report` and then ` yarn test:report:e2e -- -- -- -u` to update report snapshots.

Thanks for putting in all this effort, you're almost there!

Reply:
I found a way to make it work (I believe) but am not convinced it's the most elegant solution. 

From my understanding, the title-container styling in the files you refered (tab-stops-failed-instance-section.scss, result-section.scss, automated-checks-report.scss, and collapsible-url-result-section.scss) are not applied in the case of the report-collapsible-container.tsx as the title-container styling is nested inside other classes in all the cases and neither of these classes are applied in the report-collapsible-container.tsx case. So I ended up appling the change to the content.scss file, which seems to work, but maybe a new file would have been better? 

And thanks for the guidance - again, again :-)

Reply:
Thank you for taking a look at that! With the content.scss changes, I'm still seeing the default `margin-block` values causing extra spacing to appear in the fastpass report:
![screenshot showing extra spacing in the updated fastpass report between the newly updated H3 and H4 elements](https://user-images.githubusercontent.com/4615491/158467126-1bfdccac-6cda-4c37-8000-f1f881b22d43.png)
In the old report, the h3 and h4 labelled elements (lowercase h indicating that these aren't native H<x> elements) are stacked with no extra spacing. In the updated report, those are now proper H3 and H4 elements, but there's now some extra spacing. 

Instead of putting the change in content.scss or the my other previous suggestions, I was able to get things looking right by adding your css changes to results-container.scss, something like:
```
.results-container {
    margin-top: 56px;

    .title-container {
        font-weight: unset;
        margin-block: 0;
        margin-inline: 0;
    }
}
```